tests/test_patterns.py

```diff
@@ -180,3 +180,17 @@ def get_kwargs(file_pattern):
         return kwargs
 
     assert get_kwargs(fp) == get_kwargs(fp_pruned)
+
+
+@pytest.mark.parametrize("file_type_value", [ft.value for ft in list(FileType)] + ["unsupported"])
+def test_setting_file_types(file_type_value):
+
+    file_type_kwargs = {"file_type": file_type_value}
+
+    if not file_type_value == "unsupported":
+        fp = make_concat_merge_pattern(**file_type_kwargs)[0]
+        assert fp.file_type == FileType(file_type_value)
+    else:
+        with pytest.raises(ValueError) as excinfo:
+            fp = make_concat_merge_pattern(**file_type_kwargs)[0]
+            assert f"'{file_type_value}' is not a valid FileType" in str(excinfo.value)
```

Comment:
I'm not sure this assertion ever gets hit. You want to use the `match=` option in `pytest.raises` to check the error message.

Reply:
https://github.com/pangeo-forge/pangeo-forge-recipes/pull/322/commits/17f47d8d7b6a633b78ebae782576470c23ed2990